test/builtin_dataset_mocks.py

```diff
@@ -430,26 +433,23 @@ def caltech256(info, root, config):
     return num_images_per_category * len(info.categories)
 
 
-@register_mock
-def imagenet(info, root, config):
+def imagenet_mock_data_fn(info, root, **options):
```

Comment:
Do we need to pass the `info`s, or could we just access/build the info dict from the within the mock?

Reply:
Like doing `info = datasets.info("imagenet")` here? I'm fine with that. I mostly left it in for compatibility with the old approach.